### ISSUE TITLE
Different usage output between help command and flag

### ISSUE THREAD

**Issue by dkalleg** [spf13/cobra#302]
I notice I'm seeing a different usage output in two scenarios.  Say I have command foo and subcommand api:

```
$foo api -h
Usage:
  foo api <endpoint>
```

This is expected since I have no flags created for the api subcommand.  However, when I use different syntax I see a different result:

```
$foo help api
Usage:
  foo api <endpoint> [flags]
```

The [flags] gets appended only in the `foo help api` scenario.  Is there a way to standardize this?

Also, fyi, my cobra command is setup expecting to NOT explicitly see [flags], like so:

```
    apiCommand  = &cobra.Command{
        Use:   "api <endpoint>",
                ...
         }
```

**Comment by traveller42** [comment]
Are there any global flags that could be used with `foo api`?

**Comment by dnephin** [comment]
I ran into this as well, it was caused the help flag being auto-added when the `help` command was run, but not when the `-h` flag was used.

I believe it was fixed by #286 which was merged a couple days ago.

**Comment by traveller42** [comment]
Thanks.

That looks like it should address this issue as well.  I'll leave that assessment to the OP.
